config/configmapprovider/yaml.go

```diff
@@ -0,0 +1,58 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configmapprovider // import "go.opentelemetry.io/collector/config/configmapprovider"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+const bytesSchemeName = "yaml"
+
+type yamlMapProvider struct{}
+
+// NewYaml returns a new Provider that allows to provide yaml bytes.
+//
+// This Provider supports "yaml" scheme, and can be called with a "location" that follows:
+//   bytes-location   = "yaml:" yaml-bytes
+//
+// Examples:
+// `yaml:processors::batch::timeout: 2s`
+// `yaml:processors::batch/foo::timeout: 3s`
+func NewYaml() Provider {
```

Comment:
Shouldn't this be `NewYAML` per Go conventions?

Reply:
I wish this was a lint error.

Reply:
Done.